diff_cover/report_generator.py

```diff
@@ -218,7 +218,11 @@ def _src_path_stats(self, src_path):
 
 class JsonReportGenerator(BaseReportGenerator):
     def generate_report(self, output_file):
-        json.dump(self.report_dict(), output_file)
+        json_report_str = json.dumps(self.report_dict())
+
+        # all report generators are expected to write raw bytes, so we encode
+        # the json
+        output_file.write(json_report_str.encode())
```

Comment:
I would prefer we were explicit with the encoding. 'utf-8' ?

Reply:
(I know in python 3 this defaults to utf-8, but in 2 it does not)

Reply:
Could I ask why Python 2 support hasn't been dropped?

Reply:
Coverage tools typically support a wider range of python versions than libraries and the like and for poor souls who have not migrated to 3 yet bulking up their test coverage is likely a critical first step. So as long as the support burden remains small I was intending to hold on to it. 

If someone were to propose a feature or a significant helpful refactoring that would would benefit from python 3 features I would not get in the way by demanding python 2 support. 

Also if coverage.py dropped 2.7 I would follow Ned's lead there.

I am the guy who ported this code to 3 back before I was made the owner of it after all.